From 77d8eda1d5b80cf56147ac28ec21bdb84f54afb2 Mon Sep 17 00:00:00 2001
From: Sivashankari Madhavan <quic_sivamadh@quicinc.com>
Date: Tue, 24 Jan 2023 14:34:08 +0530
Subject: [PATCH] hostapd: Fix the station bandwidth inconsistency issue

While associating the ath12k station with AP, STA getting
associating with incorrect bandwidth. From checking, In
the beacon template VHT capability is not updated with
proper capability info.

Fix it by updating the VHT capabilities.

Signed-off-by: Sivashankari Madhavan <quic_sivamadh@quicinc.com>
---
 src/ap/ieee802_11_vht.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/ap/ieee802_11_vht.c b/src/ap/ieee802_11_vht.c
index b53f521..af5b4f6 100644
--- a/src/ap/ieee802_11_vht.c
+++ b/src/ap/ieee802_11_vht.c
@@ -72,6 +72,11 @@ u8 * hostapd_eid_vht_capabilities(struct hostapd_data *hapd, u8 *eid, u32 nsts)
 		cap->vht_capabilities_info &= ~(host_to_le32(VHT_CAP_SUPP_CHAN_WIDTH_MASK));
 	} else {
 		cap->vht_capabilities_info &= ~VHT_CAP_EXTENDED_NSS_BW_SUPPORT_MASK;
+
+		if (chwidth == CHANWIDTH_160MHZ)
+			cap->vht_capabilities_info |= VHT_CAP_SUPP_CHAN_WIDTH_160MHZ;
+		else if (chwidth == CHANWIDTH_80P80MHZ)
+			cap->vht_capabilities_info |= VHT_CAP_SUPP_CHAN_WIDTH_160_80PLUS80MHZ;
 	}
 
 	/* Supported MCS set comes from hw */
-- 
2.17.1

